### ISSUE TITLE
PRI: Improve error message when referencing WinUI3 from net5.0 project

### ISSUE THREAD

**Issue by dotMorten** [microsoft/WindowsAppSDK#361]
**Describe the bug**
The build error you receive when incorrectly targeting `net5.0` instead of `net5.0-windows10.0.xxxxx` is very unhelpful:
```
Error	MSB4086	A numeric comparison was attempted on "$(TargetPlatformVersion)" that evaluates to "" instead of a number, in condition "'$(TargetPlatformVersion)' >= '10.0.11000.0'".	
ClassLibrary4	C:\Users\mn\.nuget\packages\microsoft.winui\3.0.0-preview3.201113.0\buildTransitive\Microsoft.ApplicationModel.Resources.PriGen.targets	350	
```

VS also shows the following tooltip error:
![image](https://user-images.githubusercontent.com/1378165/103188408-33acb300-487d-11eb-99bb-9b66005e1d05.png)

This issue was [reported by a user in the WinUI Discord channel](https://discord.com/channels/372137812037730304/671870147354427422/792561330984321064), and didn't know what to do at first. A better error message would have saved the person a lot of time.

**Steps to reproduce the bug**
1. Create a new .NET Core class library
2. Change Target Framework to `net5.0`
3. Add WinUI3 package
4. Compile

`.csproj`:
```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net5.0</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.WinUI" Version="3.0.0-preview3.201113.0" />
  </ItemGroup>
</Project>

```

**Expected behavior**
A helpful error that informs the user that they need to retarget.

As an example, see the PR I did for RX.NET here to help users target the correct thing: https://github.com/dotnet/reactive/pull/1442
(just put the .targets file in the `net5.0` folder instead of net5.0-windows`

**Version Info**

NuGet package version: 
[Microsoft.WinUI 3.0.0-preview3.201113.0]

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
| No | No |

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| May 2020 Update (19041)            |    |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            |  |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Additional context**

The main problem is probably that the .targets files aren't in the proper `\build\net5.0-windows10.[minversion]` folder, so it gets included in every target framework. Once that has moved to a subfolder, the error should go away (as the targets aren't included), and the approach used in the PR referenced above can be used to improve the error message.

**Comment by stevenbrix** [comment]
I transferred this issue from the WinUI repo since this is an issue with the PRI gen targets. I agree that a better error message telling the user what they need to do would be better.

**Comment by riverar** [comment]
Wanted to add I also hit this with a WPF desktop application project (net5).

**Comment by janseris** [comment]
Any news? I encountered this error today and I don't know what to do.

**Comment by dotMorten** [comment]
@janseris this really shouldn’t be an issue with Maui. Are you not multitargeting your Maui app?

**Comment by janseris** [comment]
> @janseris this really shouldn’t be an issue with Maui. Are you not multitargeting your Maui app?

I am taking this project: https://github.com/jasho/cookbook-maui
And trying to run on Windows Machine.
No modifications to the project or source code are done.
And this appears:
![image](https://user-images.githubusercontent.com/64279914/162592584-8fa469f2-28c0-4bdb-bdc2-1aedf9e1f655.png)

**Comment by dotMorten** [comment]
@janseris ? That’s a different error than what I reported here

**Comment by janseris** [comment]
> @janseris ? That’s a different error than what I reported here

> A numeric comparison was attempted on "$(TargetPlatformVersion)" that evaluates to "" instead of a number, in condition "'$(TargetPlatformIdentifier)' == 'Windows' and '$(TargetPlatformVersion)' >= '8.0'".

>A numeric comparison was attempted on "$(TargetPlatformVersion)" that evaluates to "" instead of a number , in condition "'$(TargetPlatformVersion)' >= '10.0.11000.0'".

I thought these two shared the Platform version number parsing issue.

**Comment by tubaplayerdis** [comment]
For me fixing this issue was in the line with <TargetFramework>net6.0-windows</TargetFramework>.
For me the application was looking for a target verion and i had to put that in my <TargetFramework> like this:
<TargetFramework>net6.0-windows10.0.19041.0</TargetFramework>

**Comment by elbrandt** [comment]
Hi, I'm very interested in the current status of this problem as I am experiencing the same thing on VS2022 17.2.6.
I have a WinUI3 application that builds fine with .NET6 for a Windows10 target. The project was generated with the 'Template Studio for WinUI (C#) version 5.1

When I change the Target OS (in project properties) to macOS, then I get the shown below
![image](https://user-images.githubusercontent.com/36109794/179813766-286edc96-8112-4c1e-98cc-b1dfd984169d.png)
:

The reason is that the line in the csproj file:
<TargetFramework>net6.0-windows10.0.19041.0</TargetFramework>
gets changed to 
<TargetFramework>net6.0-macos</TargetFramework>

A google search shows this error popping up for people in various different scenarios, but in my case, I want to be able to build a WinUI app for macOS. Google searches never turn up a satisfactory explanation or solution (maybe because things are changing so fast...)

Advice? Thanks!

**Comment by hdufortlaumie** [comment]
I've a similar issue with WinUI and dotnet 7. 
I've create a new WinUI3 project , flipping starget Framework to .Net 7 instead of .Net 6. raise this error:

_2022-11-16 15:49:20	Critical	Project 'App1' load failed|   [MSB4086] A numeric comparison was attempted on "$(TargetPlatformVersion)" that evaluates to "" instead of a number, in condition "'$(TargetPlatformVersion)' < '10.0.18362.0'".  C:\Users\hugod\.nuget\packages\microsoft.windowsappsdk\1.2.221109.1\buildTransitive\Microsoft.InteractiveExperiences.Common.targets at (15:27)_

I'm using Visual Studio Pro v17.4.1

**Comment by mazen428** [comment]
Same, switching target framework to dotnet 7 does not work even though it's supposed to be supported.

**Comment by francodelucchi** [comment]
> I've a similar issue with WinUI and dotnet 7. I've create a new WinUI3 project , flipping starget Framework to .Net 7 instead of .Net 6. raise this error:
> 
> _2022-11-16 15:49:20 Critical Project 'App1' load failed| [MSB4086] A numeric comparison was attempted on "$(TargetPlatformVersion)" that evaluates to "" instead of a number, in condition "'$(TargetPlatformVersion)' < '10.0.18362.0'". C:\Users\hugod.nuget\packages\microsoft.windowsappsdk\1.2.221109.1\buildTransitive\Microsoft.InteractiveExperiences.Common.targets at (15:27)_
> 
> I'm using Visual Studio Pro v17.4.1



> For me fixing this issue was in the line with net6.0-windows. For me the application was looking for a target version and i had to put that in my like this: net6.0-windows10.0.19041.0

You saved me with this post, thanks! :-) Exactly the same problem, fixed this way!

**Comment by JohnGomes** [comment]
> For me fixing this issue was in the line with net6.0-windows. For me the application was looking for a target verion and i had to put that in my like this: net6.0-windows10.0.19041.0

Yup, I had to follow a similar formula -- net{VersionNumber}.0-windows{TargetPlatformMinVersion}
net7.0-windows10.0.17763.0
![image](https://user-images.githubusercontent.com/7169199/208201546-c7c4ed60-2a5a-4c4d-9d47-5f78573a6252.png)

**Comment by AnthonyAlexandre** [comment]
JohnGomes... you saved the day... It worked on my Windows App SDK. Now is asking to upgrade some dependency class libraries... i wil do anyways... :)

**Comment by GamerGirlandCo** [comment]
For me, right clicking the faulted project and selecting `Reload project with dependencies` (NOT simply `Reload Project`) worked
